crt_portal/cts_forms/views.py

```diff
@@ -182,6 +183,38 @@ def _related_reports_count(report):
         return 1
 
 
+def _get_email(report):
+    return report.contact_email
+
+
+def _set_constant_writer(contact_templates, report):
+    for contact_template in contact_templates:
+        if contact_template[1].upper() == report.contact_email.upper():
+            report.constant_writer = True
+
+
+def _was_sent_constant_writer_email(reports):
+    report_emails = tuple(map(_get_email, reports))
+    if len(report_emails) == 0:
+        return
+    with connection.cursor() as cursor:
+        try:
+            cursor.execute(
+                """  select act.description as description, r.contact_email as email
+                from cts_forms_report r
+                left join actstream_action act on concat('Email sent: ''CRT - Constant Writer'' to ', r.contact_email, ' via govDelivery TMS') = act.description
+                WHERE act.verb = 'Contacted complainant:'
+                AND r.contact_email IN %s
+                order by act.timestamp desc
```

Comment:
SQL readability nit: be consistent about upper vs lowercase keywords. I personally prefer uppercase.

```suggestion
                """  SELECT act.description AS description, r.contact_email AS email
                FROM cts_forms_report r
                LEFT JOIN actstream_action act ON CONCAT('Email sent: ''CRT - Constant Writer'' to ', r.contact_email, ' via govDelivery TMS') = act.description
                WHERE act.verb = 'Contacted complainant:'
                AND r.contact_email IN %s
                ORDER BY act.timestamp DESC
```